src/lib/optimizer.js

```diff
@@ -111,7 +111,7 @@ export const Optimizer = {
       }
     }
 
-    let relics = Utils.clone(DB.getRelics());
+    let relics = Utils.clone(DB.getFilteredRelics());
```

Comment:
@fribbels DB.getRelics() seems to grab the full set here.
e.g., Permutations count would show "250,000", but optimizing would result in millions of hits (using my personal kel-z import data).

had to give explicit get/set to get optimizer to correctly calc with filtered out relics.  maybe a bug from before?  or did i miss the schema/filter logic completely somewhere else?

Reply:
@cnojima 

This was duplicated code that I've been meaning to refactor - but I think the issue this introduces is that if you set filters -> calculate permutations -> save filtered relics -> but then edit a relic / import new relics -> the optimizer uses the cached version of the old relics not the updated ones. 

The filter methods in this file will get cleaned up into RelicFilters so we don't have this duplication and then maybe we can just called RelicFilters.applyAllFilters() in one place.

For now I think we can add a applyEquippedFilter to this duplicated filter list to resolve your 250,000 vs 2m discrepancy

Reply:
https://github.com/fribbels/hsr-optimizer/issues/55